learning_assistant/views.py

```diff
@@ -0,0 +1,100 @@
+"""
+V1 API Views.
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status as http_status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+try:
+    from common.djangoapps.course_modes.models import CourseMode
+    from common.djangoapps.student.models import CourseEnrollment
+    from lms.djangoapps.courseware.access import get_user_role
+    from lms.djangoapps.courseware.toggles import learning_assistant_is_active
+except ImportError:
+    # If the waffle flag is false, the endpoint will force an early return.
+    learning_assistant_is_active = False
+
+from learning_assistant.models import CoursePrompt
+from learning_assistant.serializers import MessageSerializer
+from learning_assistant.utils import get_chat_response
+
+log = logging.getLogger(__name__)
+
+
+class CourseChatView(APIView):
+    """
+    View to retrieve chat response.
+    """
+
+    authentication_classes = (SessionAuthentication, JwtAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, course_id):
+        """
+        Given a course ID, retrieve a chat response for that course.
+
+        Expected POST data: {
+            [
+                {'role': 'user', 'content': 'What is 2+2?'},
+                {'role': 'assistant', 'content': '4'}
+            ]
+        }
+        """
+        if not learning_assistant_is_active(course_id):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Learning assistant not enabled for course.'}
+            )
+
+        # get user does not have a verified enrollment, or is not staff, they should not have access
+        user_role = get_user_role(request.user, course_id)
+        enrollment_object = CourseEnrollment.get_enrollment(request.user, course_id)
+        enrollment_mode = enrollment_object.mode if enrollment_object else None
+        if (
+            (enrollment_mode not in CourseMode.VERIFIED_MODES)
+            and user_role not in ('staff', 'instructor')
+        ):
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Must be staff or have valid enrollment.'}
+            )
+
+        prompt_text = CoursePrompt.get_prompt_by_course_id(course_id)
```

Comment:
Ideally, I think we would want to have a Python API module to avoid directly interfacing with the model class. But we can add that later.

Reply:
I believe that's true in the case of actually modifying the object. I feel comfortable importing the model in the case that it's only used to retrieve data

Reply:
I don't think that's true, based on our [OEP](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0049-django-app-patterns.html#api-py).

> An app’s Django models and other internal data structures should not be exposed via its Python APIs (unless performance requires it).

If we shouldn't expose or export a model from a Python API, I don't think we should import a model directly either. I don't think we want to expose the model structure or the functions of the model because it makes it harder to maintain a consistent, well-supported interface for reading from and writing to the model, particularly if we want to make changes to the model or its functions.

Let me know if you read that differently, but it's definitely up to you!

Reply:
I see what you're saying, I can update to move this to an API